tutorials/7-cosmjs/5-create-custom.md

```diff
@@ -31,120 +31,38 @@ This section assumes that you have a working Cosmos blockchain with its own modu
 
 ### Preparation
 
 This exercise assumes that:
 
-1. Your Protobuf definition files are in `./proto/myChain`.
-2. You want to compile them into TypeScript in `./client/src/types/generated`.
+1. The library you're creating is in 'myLib'.
+2. Your Protobuf definition files are in `./proto/`.
+3. You want to transpile them into TypeScript in `./src/codegen/`.
 
-Install `protoc` on your computer and its Typescript plugin in your project, possibly with the help of a Dockerfile:
-
-<CodeGroup>
-
-<CodeGroupItem title="Local">
+Install `telescope` under your package:
 
 ```sh
-$ mkdir -p /usr/lib/protoc
-$ cd /usr/lib/protoc
-$ curl -L https://github.com/protocolbuffers/protobuf/releases/download/v21.7/protoc-21.7-linux-x86_64.zip -o protoc.zip
-$ unzip -o protoc.zip
-$ rm protoc.zip
-$ ln -s /usr/lib/protoc/bin/protoc /usr/local/bin/protoc
-$ npm install ts-proto@1.121.6 --save-dev
-```
-
-Adjust to your preferred version, operating system, and CPU platform. For instance, on an Apple M1 you would use `protoc-21.7-osx-aarch_64.zip`.
-
-</CodeGroupItem>
-
-<CodeGroupItem title="Dockerfile">
-
-```Dockerfile
-FROM --platform=linux node:lts-slim as base
-ARG BUILDARCH
-
-ENV PROTOC_VERSION=21.7
-ENV TS_PROTO_VERSION=1.121.6
-
-FROM base AS platform-amd64
-ENV PROTOC_PLATFORM=x86_64
-
-FROM base AS platform-arm64
-ENV PROTOC_PLATFORM=aarch_64
-
-FROM platform-${BUILDARCH}
-
-RUN apt-get update
-RUN apt-get install curl unzip --yes
-
-# Install ProtoC
-RUN mkdir -p /usr/lib/protoc
-WORKDIR /usr/lib/protoc
-RUN curl -L https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-${PROTOC_PLATFORM}.zip -o protoc.zip
-RUN unzip -o protoc.zip
-RUN rm protoc.zip
-RUN ln -s /usr/lib/protoc/bin/protoc /usr/local/bin/protoc
-
-# Install ts-proto
-RUN npm install --global ts-proto@${TS_PROTO_VERSION} --save-exact
-
-WORKDIR /
-
-ENTRYPOINT [ "protoc" ]
+$ npm install --save-dev --save-exact @cosmology/telescope@1.0.4
 ```
 
-Then build the image:
-
-```sh
-$ docker build . -t ts-protoc
-```
-
-</CodeGroupItem>
-
-</CodeGroup>
-
 ---
 
-You can confirm the version you received. The executable is located in `./node_modules/protoc/protoc/bin/protoc`:
-
-<CodeGroup>
-
-<CodeGroupItem title="Local">
+You can confirm the version you received by running:
 
 ```sh
-$ protoc --version
+$ npx telescope --version
 ```
 
-</CodeGroupItem>
-
-<CodeGroupItem title="Docker">
-
-```sh
-$ docker run --rm -it \
-    ts-protoc --version
-```
-
-</CodeGroupItem>
-
-</CodeGroup>
-
 This returns something like:
 
 ```txt
-libprotoc 3.21.7
+1.0.4
 ```
 
 The compiler tools are ready. Time to use them.
 
-Create the target folder if it does not exist yet:
-
-```sh
-$ mkdir -p client/src/types/generated
```

Comment:
Is the folder created automatically?

Reply:
Yes, outPath will be created by Telescope itself

Reply:
Maybe we can mention that as part of the notes.

Perhaps as a troubleshoot...In particular, auto-creation of folders usually comes with silent errors, i.e. the path was wrongly written, the command completed but the user has no idea where everything went...

Reply:
This's a very good point, I've tried to make it up.